Add support for Swift Package Manager

pyrtsa opened the pull request:
This change should allow using UserReport with SPM instead of CocoaPods.

To test, create a new iOS test project and add UserReport in "File → Swift Packages → Add Package Dependency…".

pyrtsa committed:
Add support for Swift Package Manager

pyrtsa commented on Package.swift:
```diff
@@ -0,0 +1,31 @@
+            name: "UserReportSDK",
+            dependencies: [],
+            path: "UserReport/UserReport",
+            exclude: ["UserReport/UserReport/Info.plist"]
```
Still seeing a warning from this unless I replace with just `Info.plist`, so I guess it should be relative to the above `path`:
```suggestion
            exclude: ["Info.plist"]
```

pyrtsa commented on Package.swift:
```diff
@@ -0,0 +1,31 @@
+            name: "UserReportTests",
+            dependencies: ["UserReportSDK"],
+            path: "UserReport/Tests",
+            exclude: ["UserReport/Tests/Info.plist"]
```
```suggestion
            exclude: ["Info.plist"]
```

pyrtsa committed:
Use path-relative exclude paths

hmvs commented:
Hi. Sorry for the delay. Lost notification about PR. Will merge the PR next Monday (11th).

Commit:
Renaming package name

Renaming package name to make documentation relevant. So in both ways, we will have it as UserReportSDK.

hmvs commented:
Hey @pyrtsa. We've made small changes there. Renaming it from UserReport to UserReportSDK to make naming consistent across CocoaPods and SPM. Does it make sense to you?

pyrtsa commented:
Ah, sure, I think it does.

But please note—just realised this—you have your README still suggest using `import UserReport` where `import UserReportSDK` is expected. I don't know if similar leftovers from the old module name exist elsewhere in this repo.

hmvs commented:
We've tried it out and seems documentation will be still relevant, it can be imported by `import UserReport`. Would be great if you check it on your side as well @pyrtsa. I mean the latest commit in your branch.

pyrtsa commented:
No I think you misunderstand. If it can be imported like that then it's a regression of #5.

hmvs commented:
Yep, thanks. 
As far as I understand it shouldn't be an issue. After the latest commit module named `UserReportSDK`. 
![image](https://user-images.githubusercontent.com/2255040/104333009-70251380-54f9-11eb-81e1-2ec5675d3e99.png)

pyrtsa commented:
I mean, please don't. The motivation for not aliasing `UserReport` (and naming the module `UserReportSDK` instead) is better explained in the issue, #4.

hmvs commented:
Yes, I understand your concern, and it is completely valid. However, it will be a breaking change if we do that. We have to maintain compatibility with previous versions.

hmvs commented:
Thank you for PR
